app/views/layouts/application.html.slim

```diff
@@ -8,6 +8,10 @@ html.govuk-template.app-html-class lang="en"
     title #{content_for :page_title_prefix} — #{t("app.title")}
     meta content="width=device-width, initial-scale=1" name="viewport"
     meta content="blue" name="theme-color"
+
+    - if canonical_href
+      link rel="canonical" href="#{canonical_href}"
```

Comment:
There's a [content area for links in the application layout](https://github.com/DFE-Digital/teaching-vacancies/blob/master/app/views/layouts/application.html.slim#L17). You can use something like:
```slim
- content_for(:head_links) do
  link rel="canonical" href="#{canonical_href}"
```
in `views/vacancies/show.html.slim`, that way you do not need to a) edit the layout, and b) add `canonical_href` to `ApplicationController`: https://guides.rubyonrails.org/layouts_and_rendering.html#using-the-content-for-method